src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -145,12 +146,8 @@ public void setPeriod(String period) {
     public void visitJavadocToken(DetailNode ast) {
         String firstSentence = getFirstSentence(ast);
         final int endOfSentence = firstSentence.lastIndexOf(period);
-        if (endOfSentence == -1) {
-            if (!isOnlyInheritDoc(ast)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
-            }
-        }
-        else {
+        checkPeriodAtEnd(ast);
```

Comment:
method `visitJavadocToken` should contain all `log` statements. All methods should return results to make final decision at `visitJavadocToken`. Please refactor `checkPeriodAtEnd` method to return results to let you inline your `validate` method to `visitJavadocToken`